modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WMTSHelper.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.ows.wmts;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Map;
+
+/**
+ * Some useful functions that was used within gt-wmts.
+ *
+ * <p>Might have their substitution elsewhere
+ *
+ * @author Roar Brænden
+ */
+public class WMTSHelper {
+
+    public static String replaceToken(String baseUrl, String dimName, String dimValue) {
+        String token = "{" + dimName.toLowerCase() + "}";
+        int index = baseUrl.toLowerCase().indexOf(token);
+        if (index != -1) {
+            return baseUrl.substring(0, index)
+                    + (dimValue == null ? "" : dimValue)
+                    + baseUrl.substring(index + dimName.length() + 2);
+        } else {
+            return baseUrl;
+        }
+    }
+
+    /**
+     * Append the parameters to the baseUrl.
+     *
+     * <p>Add a ? mark if baseUrl doesn't have it.
+     *
+     * <p>Does not add existing parameters
+     *
+     * @param baseUrl
```

Comment:
No point having @param and @return if not documented. I often just get rid of them (they are not mandatory)